src/Raven.Server/Documents/Indexes/Static/EsprimaVisitor.cs

```diff
@@ -0,0 +1,767 @@
+                    VisitForOfStatement(node.As<ForOfStatement>());
+                    break;
+                case Nodes.MethodDefinition:
+                    VisitMethodDefinition(node.As<MethodDefinition>());
+                    break;
+                case Nodes.ImportSpecifier:
+                    VisitImportSpecifier(node.As<ImportSpecifier>());
+                    break;
+                case Nodes.ImportDefaultSpecifier:
+                    VisitImportDefaultSpecifier(node.As<ImportDefaultSpecifier>());
+                    break;
+                case Nodes.ImportNamespaceSpecifier:
+                    VisitImportNamespaceSpecifier(node.As<ImportNamespaceSpecifier>());
+                    break;
+                case Nodes.ImportDeclaration:
+                    VisitImportDeclaration(node.As<ImportDeclaration>());
+                    break;
+                case Nodes.ExportSpecifier:
+                    VisitExportSpecifier(node.As<ExportSpecifier>());
+                    break;
+                case Nodes.ExportNamedDeclaration:
+                    VisitExportNamedDeclaration(node.As<ExportNamedDeclaration>());
+                    break;
+                case Nodes.ExportAllDeclaration:
+                    VisitExportAllDeclaration(node.As<ExportAllDeclaration>());
+                    break;
+                case Nodes.ExportDefaultDeclaration:
+                    VisitExportDefaultDeclaration(node.As<ExportDefaultDeclaration>());
+                    break;
+                case Nodes.ClassExpression:
+                    VisitClassExpression(node.As<ClassExpression>());
+                    break;
+            }
+        }
+
+        public virtual void VisitClassExpression(ClassExpression classExpression)
+        {
+        }
+
+        public virtual void VisitExportDefaultDeclaration(ExportDefaultDeclaration exportDefaultDeclaration)
+        {
+        }
+
+        public virtual void VisitExportAllDeclaration(ExportAllDeclaration exportAllDeclaration)
+        {
+        }
+
+        public virtual void VisitExportNamedDeclaration(ExportNamedDeclaration exportNamedDeclaration)
+        {
+        }
+
+        public virtual void VisitExportSpecifier(ExportSpecifier exportSpecifier)
+        {
+        }
+
+        public virtual void VisitImportDeclaration(ImportDeclaration importDeclaration)
+        {
+        }
+
+        public virtual void VisitImportNamespaceSpecifier(ImportNamespaceSpecifier importNamespaceSpecifier)
+        {
+        }
+
+        public virtual void VisitImportDefaultSpecifier(ImportDefaultSpecifier importDefaultSpecifier)
+        {
+        }
+
+        public virtual void VisitImportSpecifier(ImportSpecifier importSpecifier)
+        {
+        }
+
+        public virtual void VisitMethodDefinition(MethodDefinition methodDefinitionas)
+        {
+        }
+
+        public virtual void VisitForOfStatement(ForOfStatement forOfStatement)
+        {
```

Comment:
Missing implementation here.

Reply:
Jint doesn't seem to have support for this, it is part of the completion of the solution that i have added it, but i'll add an implementation to support it in the visitor.